quickwit/quickwit-search/src/leaf.rs

```diff
@@ -1218,33 +1218,17 @@ pub async fn multi_index_leaf_search(
                     doc_mapper,
                     aggregation_limits,
                 )
+                .in_current_span()
                 .await
             }
-            .in_current_span()
         });
-        leaf_request_tasks.push(leaf_request_future);
     }
 
-    let leaf_responses: Vec<crate::Result<LeafSearchResponse>> = tokio::time::timeout(
-        searcher_context.searcher_config.request_timeout(),
-        try_join_all(leaf_request_tasks),
-    )
-    .await??;
+    let leaf_responses: Vec<LeafSearchResponse> = try_join_vec.try_join_all().await?;
```

Comment:
I know this is how things are currently, but do we actually need the responses to come back in the same order. That seems odd.